From 24c5633c02073b63ca754c7f6e3df77619224d1a Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Tue, 12 Jun 2018 11:54:15 +0300
Subject: [PATCH 159/706] enetc: move get_si_caps after PF port init

Port init should be done first to pre-allocate rings and other resources.
PF driver should pick up its resources with get_si_caps after port init is
done.  This way PF driver gets the correct number of rings allocated to it.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
(cherry picked from commit be67dc4d512c968cf48672458d4028eef1eb29ae)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index ff4bcca73a50..eff72fd1e517 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -710,14 +710,14 @@ static int enetc_pf_probe(struct pci_dev *pdev,
 		goto err_map_pf_space;
 	}
 
-	enetc_get_si_caps(si);
-
 	pf = enetc_si_priv(si);
 	pf->si = si;
 	pf->total_vfs = pci_sriov_get_totalvfs(pdev);
 
 	enetc_configure_port(pf);
 
+	enetc_get_si_caps(si);
+
 	ndev = alloc_etherdev_mq(sizeof(*priv), ENETC_MAX_NUM_TXQS);
 	if (!ndev) {
 		err = -ENOMEM;
-- 
2.17.1

